### ISSUE TITLE
look-at bug?

### ISSUE THREAD

**Issue by MatthewRock** [cbaggers/rtg-math#30]
I tried following the tutorial: http://www.opengl-tutorial.org/beginners-tutorials/tutorial-3-matrices/

When trying to copy the matrices from the end of the tutorial (where you need to print the triangle using some matrix transformations; you can look it up), the result differs from the tutorial - it looks like look-at might be the culprit.

The ready-to-build code can be found here https://github.com/MatthewRock/opengl-learn
NOTE: the top commit on master branch (a) doesn't *really* work, and (b) does not use your library. Please go to HEAD~1 ("v3" commit), to see my tries. It may not be 1-to-1 tutorial matrix, because I've been experimenting a bit, but it will allow you to test the code without many modifications.

I won't lie by pretending that I know what's going on; I don't. However, I've seen some tutorial ( http://www.geertarien.com/blog/2017/07/30/breakdown-of-the-lookAt-function-in-OpenGL/ ) and from my brief look at the code, there are some differences. You can also compare the result of using your library with the top commit on master branch (using box.math).

**Comment by cbaggers** [comment]
Good stuff! Thanks for the report and the research that went into it. I'd like to move to this approach as, even if the current version was correct (which I don't think it is) this version will be faster (don't need that extra allocation and matrix multiply)

The version I have is just the transpose of `from-direction` function with translation tacked onto it. That function should also be reviewed.

**Comment by cbaggers** [comment]
I'll be looking at this today

**Comment by MatthewRock** [comment]
That's great!

Also, begs for a "looking at" pun.

**Comment by cbaggers** [comment]
gah puns! :p 
got distracted by an macOS cl-sdl2 issue last night. Starting on this now

**Comment by cbaggers** [comment]
Getting some unexpected results, need to look deeper into this

**Comment by cbaggers** [comment]
@MatthewRock each time I have to look at this kind of stuff again I feel happy that I've learned more since the last time but also distraught at how wrong some of my code is.

Also it's maddening when tutorials are wrong. I'm going to stick to GLM as a source of truth from now on. Turns out glm's lookat matrix is different from the one in the breakdown article (which drove me mad for hours today as I couldn't understand how it was valid) Finally I'm getting the results I expect.

In the process I found some mistakes in the projection matrices, which is fairly mortifying. Ah well this is how we learn I guess.. just wish I could have got this right the first (or second) times I tried this.

Thanks for bearing with me during this. I'll merge this later today, I'm going to do a survey of some more matrix code before I do however
